<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.8"/>
<title>RNAlib-2.2.4: Functions to Read/Write several File Formats for RNA Sequences, Structures, and Alignments</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">RNAlib-2.2.4
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.8 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="annotated.html"><span>Data&#160;Structures</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('group__file__utils.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Data Structures</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Typedefs</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Enumerations</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(7)"><span class="SelectionMark">&#160;</span>Enumerator</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(8)"><span class="SelectionMark">&#160;</span>Macros</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(9)"><span class="SelectionMark">&#160;</span>Modules</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(10)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#files">Files</a> &#124;
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">Functions to Read/Write several File Formats for RNA Sequences, Structures, and Alignments<div class="ingroups"><a class="el" href="group__utils.html">Utilities</a></div></div>  </div>
</div><!--header-->
<div class="contents">
<div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png" alt="+"/> Collaboration diagram for Functions to Read/Write several File Formats for RNA Sequences, Structures, and Alignments:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
<center><table><tr><td><div class="center"><iframe scrolling="no" frameborder="0" src="group__file__utils.svg" width="340" height="82"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</td></tr></table></center>
</div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="files"></a>
Files</h2></td></tr>
<tr class="memitem:file__formats_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="file__formats_8h.html">file_formats.h</a></td></tr>
<tr class="memdesc:file__formats_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Functions dealing with file formats for RNA sequences, structures, and alignments. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ribo_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="ribo_8h.html">ribo.h</a></td></tr>
<tr class="memdesc:ribo_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Parse RiboSum Scoring Matrices for Covariance Scoring of Alignments. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:gaaface7db12fadc3d271641c4515ab6e4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#gaaface7db12fadc3d271641c4515ab6e4">vrna_file_helixlist</a> (const char *seq, const char *db, float energy, FILE *file)</td></tr>
<tr class="memdesc:gaaface7db12fadc3d271641c4515ab6e4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Print a secondary structure as helix list.  <a href="#gaaface7db12fadc3d271641c4515ab6e4">More...</a><br /></td></tr>
<tr class="separator:gaaface7db12fadc3d271641c4515ab6e4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gab69682373ccca1e0e28cc967eec07745"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#gab69682373ccca1e0e28cc967eec07745">vrna_file_connect</a> (const char *seq, const char *db, float energy, const char *identifier, FILE *file)</td></tr>
<tr class="memdesc:gab69682373ccca1e0e28cc967eec07745"><td class="mdescLeft">&#160;</td><td class="mdescRight">Print a secondary structure as connect table.  <a href="#gab69682373ccca1e0e28cc967eec07745">More...</a><br /></td></tr>
<tr class="separator:gab69682373ccca1e0e28cc967eec07745"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga9b462e6f202594af5d3fa56e280d633f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#ga9b462e6f202594af5d3fa56e280d633f">vrna_file_bpseq</a> (const char *seq, const char *db, FILE *file)</td></tr>
<tr class="memdesc:ga9b462e6f202594af5d3fa56e280d633f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Print a secondary structure in bpseq format.  <a href="#ga9b462e6f202594af5d3fa56e280d633f">More...</a><br /></td></tr>
<tr class="separator:ga9b462e6f202594af5d3fa56e280d633f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga31f4a6c2ea1495a6e4f9eb45a9f6193d"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#ga31f4a6c2ea1495a6e4f9eb45a9f6193d">vrna_file_json</a> (const char *seq, const char *db, double energy, const char *identifier, FILE *file)</td></tr>
<tr class="memdesc:ga31f4a6c2ea1495a6e4f9eb45a9f6193d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Print a secondary structure in jsonformat.  <a href="#ga31f4a6c2ea1495a6e4f9eb45a9f6193d">More...</a><br /></td></tr>
<tr class="separator:ga31f4a6c2ea1495a6e4f9eb45a9f6193d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga8cfb7e271efc9e1f34640acb85475639"><td class="memItemLeft" align="right" valign="top">unsigned int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#ga8cfb7e271efc9e1f34640acb85475639">vrna_file_fasta_read_record</a> (char **header, char **sequence, char ***rest, FILE *file, unsigned int options)</td></tr>
<tr class="memdesc:ga8cfb7e271efc9e1f34640acb85475639"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a (fasta) data set from a file or stdin.  <a href="#ga8cfb7e271efc9e1f34640acb85475639">More...</a><br /></td></tr>
<tr class="separator:ga8cfb7e271efc9e1f34640acb85475639"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga55a9ae6dfeecc1b3f0c2acf6fa796c15"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#ga55a9ae6dfeecc1b3f0c2acf6fa796c15">vrna_extract_record_rest_constraint</a> (char **cstruc, const char **lines, unsigned int option)</td></tr>
<tr class="memdesc:ga55a9ae6dfeecc1b3f0c2acf6fa796c15"><td class="mdescLeft">&#160;</td><td class="mdescRight">Extract a hard constraint encoded as pseudo dot-bracket string.  <a href="#ga55a9ae6dfeecc1b3f0c2acf6fa796c15">More...</a><br /></td></tr>
<tr class="separator:ga55a9ae6dfeecc1b3f0c2acf6fa796c15"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga646ebf45450a69a7f2533f9ecd283a32"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#ga646ebf45450a69a7f2533f9ecd283a32">vrna_file_SHAPE_read</a> (const char *file_name, int length, double default_value, char *sequence, double *values)</td></tr>
<tr class="memdesc:ga646ebf45450a69a7f2533f9ecd283a32"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read data from a given SHAPE reactivity input file.  <a href="#ga646ebf45450a69a7f2533f9ecd283a32">More...</a><br /></td></tr>
<tr class="separator:ga646ebf45450a69a7f2533f9ecd283a32"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gae33323c53765ecbbc410d9de2d495432"><td class="memItemLeft" align="right" valign="top"><a class="el" href="group__data__structures.html#ga8e4eb5e1bfc95776559575beb359af87">vrna_plist_t</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#gae33323c53765ecbbc410d9de2d495432">vrna_file_constraints_read</a> (const char *filename, unsigned int length, unsigned int options)</td></tr>
<tr class="memdesc:gae33323c53765ecbbc410d9de2d495432"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read constraints from an input file.  <a href="#gae33323c53765ecbbc410d9de2d495432">More...</a><br /></td></tr>
<tr class="separator:gae33323c53765ecbbc410d9de2d495432"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gafd194a69af9d92b5b0412a7627ac1595"><td class="memItemLeft" align="right" valign="top">unsigned int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#gafd194a69af9d92b5b0412a7627ac1595">read_record</a> (char **header, char **sequence, char ***rest, unsigned int options)</td></tr>
<tr class="memdesc:gafd194a69af9d92b5b0412a7627ac1595"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a data record from stdin.  <a href="#gafd194a69af9d92b5b0412a7627ac1595">More...</a><br /></td></tr>
<tr class="separator:gafd194a69af9d92b5b0412a7627ac1595"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga5e125c9586fcd4e2e1559fe76f7289cc"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga5e125c9586fcd4e2e1559fe76f7289cc"></a>
float **&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__file__utils.html#ga5e125c9586fcd4e2e1559fe76f7289cc">readribosum</a> (char *name)</td></tr>
<tr class="memdesc:ga5e125c9586fcd4e2e1559fe76f7289cc"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read a RiboSum or other user-defined Scoring Matrix and Store into global Memory. <br /></td></tr>
<tr class="separator:ga5e125c9586fcd4e2e1559fe76f7289cc"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<h2 class="groupheader">Function Documentation</h2>
<a class="anchor" id="gaaface7db12fadc3d271641c4515ab6e4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_file_helixlist </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>db</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">float&#160;</td>
          <td class="paramname"><em>energy</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">FILE *&#160;</td>
          <td class="paramname"><em>file</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Print a secondary structure as helix list. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">seq</td><td>The RNA sequence </td></tr>
    <tr><td class="paramname">db</td><td>The structure in dot-bracket format </td></tr>
    <tr><td class="paramname">file</td><td>The file handle used to print to (print defaults to 'stdout' if(file == NULL) ) </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="gab69682373ccca1e0e28cc967eec07745"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_file_connect </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>db</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">float&#160;</td>
          <td class="paramname"><em>energy</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>identifier</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">FILE *&#160;</td>
          <td class="paramname"><em>file</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Print a secondary structure as connect table. </p>
<p>Connect table file format looks like this: </p><pre class="fragment">300  ENERGY = 7.0  example
  1 G       0    2   22    1
  2 G       1    3   21    2
</pre><p> where the headerline is followed by 6 columns with:</p><ol type="1">
<li>Base number: index n</li>
<li>Base (A, C, G, T, U, X)</li>
<li>Index n-1 (0 if first nucleotide)</li>
<li>Index n+1 (0 if last nucleotide)</li>
<li>Number of the base to which n is paired. No pairing is indicated by 0 (zero).</li>
<li>Natural numbering.</li>
</ol>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">seq</td><td>The RNA sequence </td></tr>
    <tr><td class="paramname">db</td><td>The structure in dot-bracket format </td></tr>
    <tr><td class="paramname">energy</td><td>The free energy of the structure </td></tr>
    <tr><td class="paramname">identifier</td><td>An optional identifier for the sequence </td></tr>
    <tr><td class="paramname">file</td><td>The file handle used to print to (print defaults to 'stdout' if(file == NULL) ) </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga9b462e6f202594af5d3fa56e280d633f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_file_bpseq </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>db</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">FILE *&#160;</td>
          <td class="paramname"><em>file</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Print a secondary structure in bpseq format. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">seq</td><td>The RNA sequence </td></tr>
    <tr><td class="paramname">db</td><td>The structure in dot-bracket format </td></tr>
    <tr><td class="paramname">file</td><td>The file handle used to print to (print defaults to 'stdout' if(file == NULL) ) </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga31f4a6c2ea1495a6e4f9eb45a9f6193d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_file_json </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>db</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>energy</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>identifier</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">FILE *&#160;</td>
          <td class="paramname"><em>file</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Print a secondary structure in jsonformat. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">seq</td><td>The RNA sequence </td></tr>
    <tr><td class="paramname">db</td><td>The structure in dot-bracket format </td></tr>
    <tr><td class="paramname">energy</td><td>The free energy </td></tr>
    <tr><td class="paramname">identifier</td><td>An identifier for the sequence </td></tr>
    <tr><td class="paramname">file</td><td>The file handle used to print to (print defaults to 'stdout' if(file == NULL) ) </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga8cfb7e271efc9e1f34640acb85475639"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">unsigned int vrna_file_fasta_read_record </td>
          <td>(</td>
          <td class="paramtype">char **&#160;</td>
          <td class="paramname"><em>header</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char **&#160;</td>
          <td class="paramname"><em>sequence</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char ***&#160;</td>
          <td class="paramname"><em>rest</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">FILE *&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Get a (fasta) data set from a file or stdin. </p>
<p>This function may be used to obtain complete datasets from a filehandle or stdin. A dataset is always defined to contain at least a sequence. If data starts with a fasta header, i.e. a line like </p><pre class="fragment">&gt;some header info </pre><p> then <a class="el" href="group__file__utils.html#ga8cfb7e271efc9e1f34640acb85475639" title="Get a (fasta) data set from a file or stdin. ">vrna_file_fasta_read_record()</a> will assume that the sequence that follows the header may span over several lines. To disable this behavior and to assign a single line to the argument 'sequence' one can pass <a class="el" href="group__utils.html#ga0de536599b881c787b0943a2671da476" title="Input switch for vrna_file_fasta_read_record(): &quot;never allow data to span more than one line&quot;...">VRNA_INPUT_NO_SPAN</a> in the 'options' argument. If no fasta header is read in the beginning of a data block, a sequence must not span over multiple lines!<br />
 Unless the options <a class="el" href="group__utils.html#ga0f6311f11bed1842e3a527ab27b294c6" title="Input switch for get_input_line(): &quot;do not skip comment lines&quot;. ">VRNA_INPUT_NOSKIP_COMMENTS</a> or <a class="el" href="group__utils.html#gab4db885222b3b69608310d7c7e63e286" title="Input switch for vrna_file_fasta_read_record(): &quot;do not skip empty lines&quot;. ">VRNA_INPUT_NOSKIP_BLANK_LINES</a> are passed, a sequence may be interrupted by lines starting with a comment character or empty lines.<br />
 A sequence is regarded as completely read if it was either assumed to not span over multiple lines, a secondary structure or structure constraint follows the sequence on the next line, or a new header marks the beginning of a new sequence...<br />
 All lines following the sequence (this includes comments) that do not initiate a new dataset according to the above definition are available through the line-array 'rest'. Here one can usually find the structure constraint or other information belonging to the current dataset. Filling of 'rest' may be prevented by passing <a class="el" href="group__utils.html#ga7a2e8c50a0c7ce82e60da1016e1367fd" title="Input switch for vrna_file_fasta_read_record(): &quot;do fill rest array&quot;. ">VRNA_INPUT_NO_REST</a> to the options argument.<br />
 </p><dl class="section note"><dt>Note</dt><dd>This function will exit any program with an error message if no sequence could be read! </dd>
<dd>
This function is NOT threadsafe! It uses a global variable to store information about the next data block.</dd></dl>
<p>The main purpose of this function is to be able to easily parse blocks of data in the header of a loop where all calculations for the appropriate data is done inside the loop. The loop may be then left on certain return values, e.g.: </p><div class="fragment"><div class="line"><a name="l00001"></a><span class="lineno">    1</span>&#160;char *id, *seq, **rest;</div>
<div class="line"><a name="l00002"></a><span class="lineno">    2</span>&#160;int  i;</div>
<div class="line"><a name="l00003"></a><span class="lineno">    3</span>&#160;id = seq = NULL;</div>
<div class="line"><a name="l00004"></a><span class="lineno">    4</span>&#160;rest = NULL;</div>
<div class="line"><a name="l00005"></a><span class="lineno">    5</span>&#160;while(!(vrna_file_fasta_read_record(&amp;id, &amp;seq, &amp;rest, NULL, 0) &amp; (VRNA_INPUT_ERROR | VRNA_INPUT_QUIT))){</div>
<div class="line"><a name="l00006"></a><span class="lineno">    6</span>&#160;  if(id) printf(&quot;%s\n&quot;, id);</div>
<div class="line"><a name="l00007"></a><span class="lineno">    7</span>&#160;  printf(&quot;%s\n&quot;, seq);</div>
<div class="line"><a name="l00008"></a><span class="lineno">    8</span>&#160;  if(rest)</div>
<div class="line"><a name="l00009"></a><span class="lineno">    9</span>&#160;    for(i=0;rest[i];i++){</div>
<div class="line"><a name="l00010"></a><span class="lineno">   10</span>&#160;      printf(&quot;%s\n&quot;, rest[i]);</div>
<div class="line"><a name="l00011"></a><span class="lineno">   11</span>&#160;      free(rest[i]);</div>
<div class="line"><a name="l00012"></a><span class="lineno">   12</span>&#160;    }</div>
<div class="line"><a name="l00013"></a><span class="lineno">   13</span>&#160;  free(rest);</div>
<div class="line"><a name="l00014"></a><span class="lineno">   14</span>&#160;  free(seq);</div>
<div class="line"><a name="l00015"></a><span class="lineno">   15</span>&#160;  free(id);</div>
<div class="line"><a name="l00016"></a><span class="lineno">   16</span>&#160;}</div>
</div><!-- fragment --><p> In the example above, the while loop will be terminated when <a class="el" href="group__file__utils.html#ga8cfb7e271efc9e1f34640acb85475639" title="Get a (fasta) data set from a file or stdin. ">vrna_file_fasta_read_record()</a> returns either an error, EOF, or a user initiated quit request.<br />
 As long as data is read from stdin (we are passing NULL as the file pointer), the id is printed if it is available for the current block of data. The sequence will be printed in any case and if some more lines belong to the current block of data each line will be printed as well.</p>
<dl class="section note"><dt>Note</dt><dd>Do not forget to free the memory occupied by header, sequence and rest!</dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">header</td><td>A pointer which will be set such that it points to the header of the record </td></tr>
    <tr><td class="paramname">sequence</td><td>A pointer which will be set such that it points to the sequence of the record </td></tr>
    <tr><td class="paramname">rest</td><td>A pointer which will be set such that it points to an array of lines which also belong to the record </td></tr>
    <tr><td class="paramname">file</td><td>A file handle to read from (if NULL, this function reads from stdin) </td></tr>
    <tr><td class="paramname">options</td><td>Some options which may be passed to alter the behavior of the function, use 0 for no options </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A flag with information about what the function actually did read </dd></dl>

</div>
</div>
<a class="anchor" id="ga55a9ae6dfeecc1b3f0c2acf6fa796c15"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_extract_record_rest_constraint </td>
          <td>(</td>
          <td class="paramtype">char **&#160;</td>
          <td class="paramname"><em>cstruc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char **&#160;</td>
          <td class="paramname"><em>lines</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>option</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Extract a hard constraint encoded as pseudo dot-bracket string. </p>
<dl class="section pre"><dt>Precondition</dt><dd>The argument 'lines' has to be a 2-dimensional character array as obtained by <a class="el" href="group__file__utils.html#ga8cfb7e271efc9e1f34640acb85475639" title="Get a (fasta) data set from a file or stdin. ">vrna_file_fasta_read_record()</a> </dd></dl>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__file__utils.html#ga8cfb7e271efc9e1f34640acb85475639" title="Get a (fasta) data set from a file or stdin. ">vrna_file_fasta_read_record()</a>, <a class="el" href="group__constraints.html#ga13053547a2de5532b64b64d35e097ae1" title="Flag that is used to indicate the pipe &#39;|&#39; sign in pseudo dot-bracket notation of hard constraints...">VRNA_CONSTRAINT_DB_PIPE</a>, <a class="el" href="group__constraints.html#ga369bea82eae75fbe626f409fa425747e" title="dot &#39;.&#39; switch for structure constraints (no constraint at all) ">VRNA_CONSTRAINT_DB_DOT</a>, <a class="el" href="group__constraints.html#ga7283bbe0f8954f7b030ecc3f2d1932b2" title="&#39;x&#39; switch for structure constraint (base must not pair) ">VRNA_CONSTRAINT_DB_X</a> <a class="el" href="group__constraints.html#gad54c1315a47d55653dcaa5de6e544b77" title="angle brackets &#39;<&#39;, &#39;>&#39; switch for structure constraint (paired downstream/upstream) ...">VRNA_CONSTRAINT_DB_ANG_BRACK</a>, <a class="el" href="group__constraints.html#gac17b034852c914bc5879954c65d7e74b" title="round brackets &#39;(&#39;,&#39;)&#39; switch for structure constraint (base i pairs base j) ">VRNA_CONSTRAINT_DB_RND_BRACK</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">cstruc</td><td>A pointer to a character array that is used as pseudo dot-bracket output </td></tr>
    <tr><td class="paramname">lines</td><td>A 2-dimensional character array with the extension lines from the FASTA input </td></tr>
    <tr><td class="paramname">option</td><td>The option flags that define the behavior and recognition pattern of this function </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga646ebf45450a69a7f2533f9ecd283a32"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int vrna_file_SHAPE_read </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>file_name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>length</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>default_value</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char *&#160;</td>
          <td class="paramname"><em>sequence</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double *&#160;</td>
          <td class="paramname"><em>values</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Read data from a given SHAPE reactivity input file. </p>
<p>This function parses the informations from a given file and stores the result in the preallocated string sequence and the double array values.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file_name</td><td>Path to the constraints file </td></tr>
    <tr><td class="paramname">length</td><td>Length of the sequence (file entries exceeding this limit will cause an error) </td></tr>
    <tr><td class="paramname">default_value</td><td>Value for missing indices </td></tr>
    <tr><td class="paramname">sequence</td><td>Pointer to an array used for storing the sequence obtained from the SHAPE reactivity file </td></tr>
    <tr><td class="paramname">values</td><td>Pointer to an array used for storing the values obtained from the SHAPE reactivity file </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="gae33323c53765ecbbc410d9de2d495432"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="group__data__structures.html#ga8e4eb5e1bfc95776559575beb359af87">vrna_plist_t</a>* vrna_file_constraints_read </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>filename</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>length</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Read constraints from an input file. </p>
<p>This function reads constraint definitions from a file and converts them into an array of <a class="el" href="group__data__structures.html#ga8e4eb5e1bfc95776559575beb359af87" title="Typename for the base pair list repesenting data structure vrna_plist_s. ">vrna_plist_t</a> data structures. The data fields of each individual returned plist entry may adopt the following configurations:</p><ul>
<li>plist.i == plist.j <img class="formulaInl" alt="$ \rightarrow $" src="form_45.png"/> single nucleotide constraint</li>
<li>plist.i != plist.j <img class="formulaInl" alt="$ \rightarrow $" src="form_45.png"/> base pair constraint</li>
<li>plist.i == 0 <img class="formulaInl" alt="$ \rightarrow $" src="form_45.png"/> End of list </li>
</ul>

</div>
</div>
<a class="anchor" id="gafd194a69af9d92b5b0412a7627ac1595"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">unsigned int read_record </td>
          <td>(</td>
          <td class="paramtype">char **&#160;</td>
          <td class="paramname"><em>header</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char **&#160;</td>
          <td class="paramname"><em>sequence</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char ***&#160;</td>
          <td class="paramname"><em>rest</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="file__formats_8h.html">ViennaRNA/file_formats.h</a>&gt;</code></p>

<p>Get a data record from stdin. </p>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000061">Deprecated:</a></b></dt><dd>This function is deprecated! Use <a class="el" href="group__file__utils.html#ga8cfb7e271efc9e1f34640acb85475639" title="Get a (fasta) data set from a file or stdin. ">vrna_file_fasta_read_record()</a> as a replacment.</dd></dl>

</div>
</div>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Fri Aug 25 2017 09:53:10 for RNAlib-2.2.4 by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.8 </li>
  </ul>
</div>
</body>
</html>
