<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.9.1"/>
<title>RNAlib-2.2.4: Soft Constraints</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { init_search(); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">RNAlib-2.2.4
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.9.1 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="annotated.html"><span>Data&#160;Structures</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('group__soft__constraints.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#groups">Modules</a> &#124;
<a href="#files">Files</a> &#124;
<a href="#nested-classes">Data Structures</a> &#124;
<a href="#define-members">Macros</a> &#124;
<a href="#typedef-members">Typedefs</a> &#124;
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">Soft Constraints<div class="ingroups"><a class="el" href="group__folding__routines.html">RNA Secondary Structure Prediction</a> &raquo; <a class="el" href="group__constraints.html">Constraining the Secondary Structure Recursions</a></div></div>  </div>
</div><!--header-->
<div class="contents">
<div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png" alt="+"/> Collaboration diagram for Soft Constraints:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
<center><table><tr><td><div class="center"><iframe scrolling="no" frameborder="0" src="group__soft__constraints.svg" width="598" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</td></tr></table></center>
</div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="groups"></a>
Modules</h2></td></tr>
<tr class="memitem:group__generalized__sc"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__generalized__sc.html">Generalized Soft Constraints</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="files"></a>
Files</h2></td></tr>
<tr class="memitem:ligand_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="ligand_8h.html">ligand.h</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr class="memitem:structvrna__sc__s"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#structvrna__sc__s">vrna_sc_s</a></td></tr>
<tr class="memdesc:structvrna__sc__s"><td class="mdescLeft">&#160;</td><td class="mdescRight">The soft constraints data structure.  <a href="group__soft__constraints.html#structvrna__sc__s">More...</a><br /></td></tr>
<tr class="separator:structvrna__sc__s"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="define-members"></a>
Macros</h2></td></tr>
<tr class="memitem:ga62aa195893d02d1a79ca94952748df36"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga62aa195893d02d1a79ca94952748df36"></a>
#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga62aa195893d02d1a79ca94952748df36">VRNA_CONSTRAINT_SOFT_MFE</a>&#160;&#160;&#160;8192U</td></tr>
<tr class="memdesc:ga62aa195893d02d1a79ca94952748df36"><td class="mdescLeft">&#160;</td><td class="mdescRight">Soft constraints flag, apply constraints for MFE calculations. <br /></td></tr>
<tr class="separator:ga62aa195893d02d1a79ca94952748df36"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga03fb5000c19b9a2082bf4ea30a543045"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga03fb5000c19b9a2082bf4ea30a543045"></a>
#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga03fb5000c19b9a2082bf4ea30a543045">VRNA_CONSTRAINT_SOFT_PF</a>&#160;&#160;&#160;16384U</td></tr>
<tr class="memdesc:ga03fb5000c19b9a2082bf4ea30a543045"><td class="mdescLeft">&#160;</td><td class="mdescRight">Soft constraints flag, apply constraints for partition function calculations. <br /></td></tr>
<tr class="separator:ga03fb5000c19b9a2082bf4ea30a543045"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga81e10993d1ae728e4e02022b33155a12"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga81e10993d1ae728e4e02022b33155a12">VRNA_OBJECTIVE_FUNCTION_QUADRATIC</a>&#160;&#160;&#160;0</td></tr>
<tr class="memdesc:ga81e10993d1ae728e4e02022b33155a12"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the sum of squared aberrations as objective function.  <a href="#ga81e10993d1ae728e4e02022b33155a12">More...</a><br /></td></tr>
<tr class="separator:ga81e10993d1ae728e4e02022b33155a12"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gac070dfb9cafaeb14d5652bd9adf0f6b1"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#gac070dfb9cafaeb14d5652bd9adf0f6b1">VRNA_OBJECTIVE_FUNCTION_ABSOLUTE</a>&#160;&#160;&#160;1</td></tr>
<tr class="memdesc:gac070dfb9cafaeb14d5652bd9adf0f6b1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the sum of absolute aberrations as objective function.  <a href="#gac070dfb9cafaeb14d5652bd9adf0f6b1">More...</a><br /></td></tr>
<tr class="separator:gac070dfb9cafaeb14d5652bd9adf0f6b1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gae5126200d80dbb282f46083fffc606bf"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gae5126200d80dbb282f46083fffc606bf"></a>
#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#gae5126200d80dbb282f46083fffc606bf">VRNA_MINIMIZER_DEFAULT</a>&#160;&#160;&#160;0</td></tr>
<tr class="memdesc:gae5126200d80dbb282f46083fffc606bf"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use a custom implementation of the gradient descent algorithm to minimize the objective function. <br /></td></tr>
<tr class="separator:gae5126200d80dbb282f46083fffc606bf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gab1d89db58e8c497795a5005f5dbc8c4a"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#gab1d89db58e8c497795a5005f5dbc8c4a">VRNA_MINIMIZER_CONJUGATE_FR</a>&#160;&#160;&#160;1</td></tr>
<tr class="memdesc:gab1d89db58e8c497795a5005f5dbc8c4a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the GNU Scientific Library implementation of the Fletcher-Reeves conjugate gradient algorithm to minimize the objective function.  <a href="#gab1d89db58e8c497795a5005f5dbc8c4a">More...</a><br /></td></tr>
<tr class="separator:gab1d89db58e8c497795a5005f5dbc8c4a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga5aaeafe1b0aa77a5cda18943ff94b02f"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga5aaeafe1b0aa77a5cda18943ff94b02f">VRNA_MINIMIZER_CONJUGATE_PR</a>&#160;&#160;&#160;2</td></tr>
<tr class="memdesc:ga5aaeafe1b0aa77a5cda18943ff94b02f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the GNU Scientific Library implementation of the Polak-Ribiere conjugate gradient algorithm to minimize the objective function.  <a href="#ga5aaeafe1b0aa77a5cda18943ff94b02f">More...</a><br /></td></tr>
<tr class="separator:ga5aaeafe1b0aa77a5cda18943ff94b02f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga9be8a702cddf58235571ace11cc41b22"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga9be8a702cddf58235571ace11cc41b22">VRNA_MINIMIZER_VECTOR_BFGS</a>&#160;&#160;&#160;3</td></tr>
<tr class="memdesc:ga9be8a702cddf58235571ace11cc41b22"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the GNU Scientific Library implementation of the vector Broyden-Fletcher-Goldfarb-Shanno algorithm to minimize the objective function.  <a href="#ga9be8a702cddf58235571ace11cc41b22">More...</a><br /></td></tr>
<tr class="separator:ga9be8a702cddf58235571ace11cc41b22"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga7b0a65c6c92fa1d8012383ba9d3dcb4f"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga7b0a65c6c92fa1d8012383ba9d3dcb4f">VRNA_MINIMIZER_VECTOR_BFGS2</a>&#160;&#160;&#160;4</td></tr>
<tr class="memdesc:ga7b0a65c6c92fa1d8012383ba9d3dcb4f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the GNU Scientific Library implementation of the vector Broyden-Fletcher-Goldfarb-Shanno algorithm to minimize the objective function.  <a href="#ga7b0a65c6c92fa1d8012383ba9d3dcb4f">More...</a><br /></td></tr>
<tr class="separator:ga7b0a65c6c92fa1d8012383ba9d3dcb4f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga9ecd2144c2ebed7533233da3986521b0"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga9ecd2144c2ebed7533233da3986521b0">VRNA_MINIMIZER_STEEPEST_DESCENT</a>&#160;&#160;&#160;5</td></tr>
<tr class="memdesc:ga9ecd2144c2ebed7533233da3986521b0"><td class="mdescLeft">&#160;</td><td class="mdescRight">Use the GNU Scientific Library implementation of the steepest descent algorithm to minimize the objective function.  <a href="#ga9ecd2144c2ebed7533233da3986521b0">More...</a><br /></td></tr>
<tr class="separator:ga9ecd2144c2ebed7533233da3986521b0"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="typedef-members"></a>
Typedefs</h2></td></tr>
<tr class="memitem:gaa715397c7afd2d2955c315512a3d571a"><td class="memItemLeft" align="right" valign="top">typedef void(*&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#gaa715397c7afd2d2955c315512a3d571a">progress_callback</a>) (int iteration, double score, double *epsilon)</td></tr>
<tr class="memdesc:gaa715397c7afd2d2955c315512a3d571a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Callback for following the progress of the minimization process.  <a href="#gaa715397c7afd2d2955c315512a3d571a">More...</a><br /></td></tr>
<tr class="separator:gaa715397c7afd2d2955c315512a3d571a"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:ga9d977a1681356778cc66dceafbe5b032"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga9d977a1681356778cc66dceafbe5b032">vrna_sc_init</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc)</td></tr>
<tr class="memdesc:ga9d977a1681356778cc66dceafbe5b032"><td class="mdescLeft">&#160;</td><td class="mdescRight">Initialize an empty soft constraints data structure within a <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a>.  <a href="#ga9d977a1681356778cc66dceafbe5b032">More...</a><br /></td></tr>
<tr class="separator:ga9d977a1681356778cc66dceafbe5b032"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga86049d4bb0ea8674cae9b6177156b184"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga86049d4bb0ea8674cae9b6177156b184">vrna_sc_add_bp</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc, const <a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> **constraints, unsigned int options)</td></tr>
<tr class="memdesc:ga86049d4bb0ea8674cae9b6177156b184"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add soft constraints for paired nucleotides.  <a href="#ga86049d4bb0ea8674cae9b6177156b184">More...</a><br /></td></tr>
<tr class="separator:ga86049d4bb0ea8674cae9b6177156b184"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga30f30c8eff9676775a3e831d972b5284"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga30f30c8eff9676775a3e831d972b5284">vrna_sc_add_up</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc, const <a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> *constraints, unsigned int options)</td></tr>
<tr class="memdesc:ga30f30c8eff9676775a3e831d972b5284"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add soft constraints for unpaired nucleotides.  <a href="#ga30f30c8eff9676775a3e831d972b5284">More...</a><br /></td></tr>
<tr class="separator:ga30f30c8eff9676775a3e831d972b5284"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga73cdc07b9a199c614367bebef0f2c41a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga73cdc07b9a199c614367bebef0f2c41a">vrna_sc_remove</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc)</td></tr>
<tr class="memdesc:ga73cdc07b9a199c614367bebef0f2c41a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Remove soft constraints from <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a>.  <a href="#ga73cdc07b9a199c614367bebef0f2c41a">More...</a><br /></td></tr>
<tr class="separator:ga73cdc07b9a199c614367bebef0f2c41a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga6d55446448d69346fc313b993c4fb3e8"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga6d55446448d69346fc313b993c4fb3e8">vrna_sc_free</a> (<a class="el" href="group__constraints.html#ga75401ce219ef8dbcceb672db82d434c6">vrna_sc_t</a> *sc)</td></tr>
<tr class="memdesc:ga6d55446448d69346fc313b993c4fb3e8"><td class="mdescLeft">&#160;</td><td class="mdescRight">Free memory occupied by a <a class="el" href="group__constraints.html#ga75401ce219ef8dbcceb672db82d434c6" title="Typename for the soft constraints data structure vrna_sc_s. ">vrna_sc_t</a> data structure.  <a href="#ga6d55446448d69346fc313b993c4fb3e8">More...</a><br /></td></tr>
<tr class="separator:ga6d55446448d69346fc313b993c4fb3e8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga57d612b58e1c61dd6cfcb5a843f8f1b3"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga57d612b58e1c61dd6cfcb5a843f8f1b3">vrna_sc_add_SHAPE_deigan</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc, const double *reactivities, double m, double b, unsigned int options)</td></tr>
<tr class="memdesc:ga57d612b58e1c61dd6cfcb5a843f8f1b3"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add SHAPE reactivity data as soft constraints (Deigan et al. method)  <a href="#ga57d612b58e1c61dd6cfcb5a843f8f1b3">More...</a><br /></td></tr>
<tr class="separator:ga57d612b58e1c61dd6cfcb5a843f8f1b3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga04ba85da63d8c793bb8001d1e6f800ba"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga04ba85da63d8c793bb8001d1e6f800ba">vrna_sc_add_SHAPE_deigan_ali</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc, const char **shape_files, const int *shape_file_association, double m, double b, unsigned int options)</td></tr>
<tr class="memdesc:ga04ba85da63d8c793bb8001d1e6f800ba"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add SHAPE reactivity data from files as soft constraints for consensus structure prediction (Deigan et al. method)  <a href="#ga04ba85da63d8c793bb8001d1e6f800ba">More...</a><br /></td></tr>
<tr class="separator:ga04ba85da63d8c793bb8001d1e6f800ba"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gaf3c65a045060aef5c4e41693d30af58c"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#gaf3c65a045060aef5c4e41693d30af58c">vrna_sc_add_SHAPE_zarringhalam</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc, const double *reactivities, double b, double default_value, const char *shape_conversion, unsigned int options)</td></tr>
<tr class="memdesc:gaf3c65a045060aef5c4e41693d30af58c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add SHAPE reactivity data as soft constraints (Zarringhalam et al. method)  <a href="#gaf3c65a045060aef5c4e41693d30af58c">More...</a><br /></td></tr>
<tr class="separator:gaf3c65a045060aef5c4e41693d30af58c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga67675b3ed48744489a3bcfa4174197cb"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ga67675b3ed48744489a3bcfa4174197cb">vrna_sc_SHAPE_to_pr</a> (const char *shape_conversion, double *values, int length, double default_value)</td></tr>
<tr class="memdesc:ga67675b3ed48744489a3bcfa4174197cb"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert SHAPE reactivity values to probabilities for being unpaired.  <a href="#ga67675b3ed48744489a3bcfa4174197cb">More...</a><br /></td></tr>
<tr class="separator:ga67675b3ed48744489a3bcfa4174197cb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gaa124bdc20d88001c38ade590c4bcc3c4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#gaa124bdc20d88001c38ade590c4bcc3c4">vrna_sc_minimize_pertubation</a> (<a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *vc, const double *q_prob_unpaired, int objective_function, double sigma_squared, double tau_squared, int algorithm, int sample_size, double *epsilon, double initialStepSize, double minStepSize, double minImprovement, double minimizerTolerance, <a class="el" href="group__soft__constraints.html#gaa715397c7afd2d2955c315512a3d571a">progress_callback</a> callback)</td></tr>
<tr class="memdesc:gaa124bdc20d88001c38ade590c4bcc3c4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find a vector of perturbation energies that minimizes the discripancies between predicted and observed pairing probabilities and the amount of neccessary adjustments.  <a href="#gaa124bdc20d88001c38ade590c4bcc3c4">More...</a><br /></td></tr>
<tr class="separator:gaa124bdc20d88001c38ade590c4bcc3c4"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<hr/><h2 class="groupheader">Data Structure Documentation</h2>
<a name="structvrna__sc__s" id="structvrna__sc__s"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">struct vrna_sc_s</td>
        </tr>
      </table>
</div><div class="memdoc">
<div class="textblock"><p>The soft constraints data structure. </p>
</div><table class="memberdecls">
<tr><td colspan="2"><h3>Data Fields</h3></td></tr>
<tr class="memitem:a57e4dbb924ab11f304e3762a3a9b07a1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a57e4dbb924ab11f304e3762a3a9b07a1"></a>
int **&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a57e4dbb924ab11f304e3762a3a9b07a1">energy_up</a></td></tr>
<tr class="memdesc:a57e4dbb924ab11f304e3762a3a9b07a1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Energy contribution for stretches of unpaired nucleotides. <br /></td></tr>
<tr class="separator:a57e4dbb924ab11f304e3762a3a9b07a1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad139b8e06632e00cbcf3909815d0d03d"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad139b8e06632e00cbcf3909815d0d03d"></a>
int *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ad139b8e06632e00cbcf3909815d0d03d">energy_bp</a></td></tr>
<tr class="memdesc:ad139b8e06632e00cbcf3909815d0d03d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Energy contribution for base pairs. <br /></td></tr>
<tr class="separator:ad139b8e06632e00cbcf3909815d0d03d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad3b4972d3b6c23865587e4ac56a37375"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad3b4972d3b6c23865587e4ac56a37375"></a>
<a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> **&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ad3b4972d3b6c23865587e4ac56a37375">exp_energy_up</a></td></tr>
<tr class="memdesc:ad3b4972d3b6c23865587e4ac56a37375"><td class="mdescLeft">&#160;</td><td class="mdescRight">Boltzmann Factors of the energy contributions for unpaired sequence stretches. <br /></td></tr>
<tr class="separator:ad3b4972d3b6c23865587e4ac56a37375"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a62c22be478cd630541c91f73e6cb0d75"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a62c22be478cd630541c91f73e6cb0d75"></a>
<a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a62c22be478cd630541c91f73e6cb0d75">exp_energy_bp</a></td></tr>
<tr class="memdesc:a62c22be478cd630541c91f73e6cb0d75"><td class="mdescLeft">&#160;</td><td class="mdescRight">Boltzmann Factors of the energy contribution for base pairs. <br /></td></tr>
<tr class="separator:a62c22be478cd630541c91f73e6cb0d75"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac20dded6068e81acd0f1139092f66a22"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac20dded6068e81acd0f1139092f66a22"></a>
int *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#ac20dded6068e81acd0f1139092f66a22">energy_stack</a></td></tr>
<tr class="memdesc:ac20dded6068e81acd0f1139092f66a22"><td class="mdescLeft">&#160;</td><td class="mdescRight">Pseudo Energy contribution per base pair involved in a stack. <br /></td></tr>
<tr class="separator:ac20dded6068e81acd0f1139092f66a22"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4a0058fe3d5ba3416f0aaab677610115"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a4a0058fe3d5ba3416f0aaab677610115"></a>
<a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a4a0058fe3d5ba3416f0aaab677610115">exp_energy_stack</a></td></tr>
<tr class="memdesc:a4a0058fe3d5ba3416f0aaab677610115"><td class="mdescLeft">&#160;</td><td class="mdescRight">Boltzmann weighted pseudo energy contribution per nucleotide involved in a stack. <br /></td></tr>
<tr class="separator:a4a0058fe3d5ba3416f0aaab677610115"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a32dc86090237888c75491bbd4861a04b"><td class="memItemLeft" align="right" valign="top"><a class="el" href="group__generalized__sc.html#gaf38062858ac25fd5e240c2c3b0b0b780">vrna_callback_sc_energy</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a32dc86090237888c75491bbd4861a04b">f</a></td></tr>
<tr class="memdesc:a32dc86090237888c75491bbd4861a04b"><td class="mdescLeft">&#160;</td><td class="mdescRight">A function pointer used for pseudo energy contribution in MFE calculations.  <a href="#a32dc86090237888c75491bbd4861a04b">More...</a><br /></td></tr>
<tr class="separator:a32dc86090237888c75491bbd4861a04b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2a2aca01782c2b980d7b7fd05b9be89c"><td class="memItemLeft" align="right" valign="top"><a class="el" href="group__generalized__sc.html#gaa216f513c3b0bd6fe5807dd0c53a8e5a">vrna_callback_sc_backtrack</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a2a2aca01782c2b980d7b7fd05b9be89c">bt</a></td></tr>
<tr class="memdesc:a2a2aca01782c2b980d7b7fd05b9be89c"><td class="mdescLeft">&#160;</td><td class="mdescRight">A function pointer used to obtain backtraced base pairs in loop regions that were altered by soft constrained pseudo energy contributions.  <a href="#a2a2aca01782c2b980d7b7fd05b9be89c">More...</a><br /></td></tr>
<tr class="separator:a2a2aca01782c2b980d7b7fd05b9be89c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0de08a09f3ccf2f97974d23192668ab0"><td class="memItemLeft" align="right" valign="top"><a class="el" href="group__generalized__sc.html#ga2eade8745c163a553763be4cfe2a679b">vrna_callback_sc_exp_energy</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a0de08a09f3ccf2f97974d23192668ab0">exp_f</a></td></tr>
<tr class="memdesc:a0de08a09f3ccf2f97974d23192668ab0"><td class="mdescLeft">&#160;</td><td class="mdescRight">A function pointer used for pseudo energy contribution boltzmann factors in PF calculations.  <a href="#a0de08a09f3ccf2f97974d23192668ab0">More...</a><br /></td></tr>
<tr class="separator:a0de08a09f3ccf2f97974d23192668ab0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7574680143df97b9029146c2150bf06d"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7574680143df97b9029146c2150bf06d"></a>
void *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__soft__constraints.html#a7574680143df97b9029146c2150bf06d">data</a></td></tr>
<tr class="memdesc:a7574680143df97b9029146c2150bf06d"><td class="mdescLeft">&#160;</td><td class="mdescRight">A pointer to the data object provided for for pseudo energy contribution functions of the generalized soft constraints feature. <br /></td></tr>
<tr class="separator:a7574680143df97b9029146c2150bf06d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<h4 class="groupheader">Field Documentation</h4>
<a class="anchor" id="a32dc86090237888c75491bbd4861a04b"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="group__generalized__sc.html#gaf38062858ac25fd5e240c2c3b0b0b780">vrna_callback_sc_energy</a>* vrna_sc_s::f</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>A function pointer used for pseudo energy contribution in MFE calculations. </p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__generalized__sc.html#ga8c7d907ec0125cd61c04e0908010a4e9" title="Bind a function pointer for generalized soft constraint feature (MFE version) ">vrna_sc_add_f()</a> </dd></dl>

</div>
</div>
<a class="anchor" id="a2a2aca01782c2b980d7b7fd05b9be89c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="group__generalized__sc.html#gaa216f513c3b0bd6fe5807dd0c53a8e5a">vrna_callback_sc_backtrack</a>* vrna_sc_s::bt</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>A function pointer used to obtain backtraced base pairs in loop regions that were altered by soft constrained pseudo energy contributions. </p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__generalized__sc.html#gabde7d07a79bb9a8f4721aee247b674ea" title="Bind a backtracking function pointer for generalized soft constraint feature. ">vrna_sc_add_bt()</a> </dd></dl>

</div>
</div>
<a class="anchor" id="a0de08a09f3ccf2f97974d23192668ab0"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="group__generalized__sc.html#ga2eade8745c163a553763be4cfe2a679b">vrna_callback_sc_exp_energy</a>* vrna_sc_s::exp_f</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>A function pointer used for pseudo energy contribution boltzmann factors in PF calculations. </p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__generalized__sc.html#ga87e382b5d0c9b7d9ce1b79c0473ff700" title="Bind a function pointer for generalized soft constraint feature (PF version) ">vrna_sc_add_exp_f()</a> </dd></dl>

</div>
</div>

</div>
</div>
<h2 class="groupheader">Macro Definition Documentation</h2>
<a class="anchor" id="ga81e10993d1ae728e4e02022b33155a12"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_OBJECTIVE_FUNCTION_QUADRATIC&#160;&#160;&#160;0</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the sum of squared aberrations as objective function. </p>
<p><img class="formulaInl" alt="$ F(\vec\epsilon) = \sum_{i = 1}^n{ \frac{\epsilon_i^2}{\tau^2} } + \sum_{i = 1}^n{ \frac{(p_i(\vec\epsilon) - q_i)^2}{\sigma^2} } \to min $" src="form_74.png"/> </p>

</div>
</div>
<a class="anchor" id="gac070dfb9cafaeb14d5652bd9adf0f6b1"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_OBJECTIVE_FUNCTION_ABSOLUTE&#160;&#160;&#160;1</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the sum of absolute aberrations as objective function. </p>
<p><img class="formulaInl" alt="$ F(\vec\epsilon) = \sum_{i = 1}^n{ \frac{|\epsilon_i|}{\tau^2} } + \sum_{i = 1}^n{ \frac{|p_i(\vec\epsilon) - q_i|}{\sigma^2} } \to min $" src="form_75.png"/> </p>

</div>
</div>
<a class="anchor" id="gab1d89db58e8c497795a5005f5dbc8c4a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_MINIMIZER_CONJUGATE_FR&#160;&#160;&#160;1</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the GNU Scientific Library implementation of the Fletcher-Reeves conjugate gradient algorithm to minimize the objective function. </p>
<p>Please note that this algorithm can only be used when the GNU Scientific Library is available on your system </p>

</div>
</div>
<a class="anchor" id="ga5aaeafe1b0aa77a5cda18943ff94b02f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_MINIMIZER_CONJUGATE_PR&#160;&#160;&#160;2</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the GNU Scientific Library implementation of the Polak-Ribiere conjugate gradient algorithm to minimize the objective function. </p>
<p>Please note that this algorithm can only be used when the GNU Scientific Library is available on your system </p>

</div>
</div>
<a class="anchor" id="ga9be8a702cddf58235571ace11cc41b22"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_MINIMIZER_VECTOR_BFGS&#160;&#160;&#160;3</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the GNU Scientific Library implementation of the vector Broyden-Fletcher-Goldfarb-Shanno algorithm to minimize the objective function. </p>
<p>Please note that this algorithm can only be used when the GNU Scientific Library is available on your system </p>

</div>
</div>
<a class="anchor" id="ga7b0a65c6c92fa1d8012383ba9d3dcb4f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_MINIMIZER_VECTOR_BFGS2&#160;&#160;&#160;4</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the GNU Scientific Library implementation of the vector Broyden-Fletcher-Goldfarb-Shanno algorithm to minimize the objective function. </p>
<p>Please note that this algorithm can only be used when the GNU Scientific Library is available on your system </p>

</div>
</div>
<a class="anchor" id="ga9ecd2144c2ebed7533233da3986521b0"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_MINIMIZER_STEEPEST_DESCENT&#160;&#160;&#160;5</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Use the GNU Scientific Library implementation of the steepest descent algorithm to minimize the objective function. </p>
<p>Please note that this algorithm can only be used when the GNU Scientific Library is available on your system </p>

</div>
</div>
<h2 class="groupheader">Typedef Documentation</h2>
<a class="anchor" id="gaa715397c7afd2d2955c315512a3d571a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">typedef void(* progress_callback) (int iteration, double score, double *epsilon)</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Callback for following the progress of the minimization process. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">iteration</td><td>The number of the current iteration </td></tr>
    <tr><td class="paramname">score</td><td>The score of the objective function </td></tr>
    <tr><td class="paramname">epsilon</td><td>The perturbation vector yielding the reported score </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Function Documentation</h2>
<a class="anchor" id="ga9d977a1681356778cc66dceafbe5b032"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_sc_init </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Initialize an empty soft constraints data structure within a <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a>. </p>
<p>This function adds a proper soft constraints data structure to the <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> data structure. If soft constraints already exist within the fold compound, they are removed.</p>
<dl class="section note"><dt>Note</dt><dd>Accepts vrna_fold_compound_t of type <a class="el" href="group__fold__compound.html#gga01a4ff86fa71deaaa5d1abbd95a1447da1608d3aa78905fc39e0d25a624ac9512">VRNA_VC_TYPE_SINGLE</a> and <a class="el" href="group__fold__compound.html#gga01a4ff86fa71deaaa5d1abbd95a1447da056345f1bcfe7cd595d1fd437c05246d">VRNA_VC_TYPE_ALIGNMENT</a></dd></dl>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__soft__constraints.html#ga86049d4bb0ea8674cae9b6177156b184" title="Add soft constraints for paired nucleotides. ">vrna_sc_add_bp()</a>, <a class="el" href="group__soft__constraints.html#ga30f30c8eff9676775a3e831d972b5284" title="Add soft constraints for unpaired nucleotides. ">vrna_sc_add_up()</a>, <a class="el" href="group__soft__constraints.html#ga57d612b58e1c61dd6cfcb5a843f8f1b3" title="Add SHAPE reactivity data as soft constraints (Deigan et al. method) ">vrna_sc_add_SHAPE_deigan()</a>, <a class="el" href="group__soft__constraints.html#gaf3c65a045060aef5c4e41693d30af58c" title="Add SHAPE reactivity data as soft constraints (Zarringhalam et al. method) ">vrna_sc_add_SHAPE_zarringhalam()</a>, <a class="el" href="group__soft__constraints.html#ga73cdc07b9a199c614367bebef0f2c41a" title="Remove soft constraints from vrna_fold_compound_t. ">vrna_sc_remove()</a>, <a class="el" href="group__generalized__sc.html#ga8c7d907ec0125cd61c04e0908010a4e9" title="Bind a function pointer for generalized soft constraint feature (MFE version) ">vrna_sc_add_f()</a>, <a class="el" href="group__generalized__sc.html#ga87e382b5d0c9b7d9ce1b79c0473ff700" title="Bind a function pointer for generalized soft constraint feature (PF version) ">vrna_sc_add_exp_f()</a>, vrna_sc_add_pre(), vrna_sc_add_post() </dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> where an empty soft constraint feature is to be added to </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga86049d4bb0ea8674cae9b6177156b184"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_sc_add_bp </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> **&#160;</td>
          <td class="paramname"><em>constraints</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Add soft constraints for paired nucleotides. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> the soft constraints are associated with </td></tr>
    <tr><td class="paramname">constraints</td><td>A two-dimensional array of pseudo free energies in <img class="formulaInl" alt="$ kcal / mol $" src="form_19.png"/> </td></tr>
    <tr><td class="paramname">options</td><td>The options flag indicating how/where to store the soft constraints </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga30f30c8eff9676775a3e831d972b5284"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_sc_add_up </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126">FLT_OR_DBL</a> *&#160;</td>
          <td class="paramname"><em>constraints</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Add soft constraints for unpaired nucleotides. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> the soft constraints are associated with </td></tr>
    <tr><td class="paramname">constraints</td><td>A vector of pseudo free energies in <img class="formulaInl" alt="$ kcal / mol $" src="form_19.png"/> </td></tr>
    <tr><td class="paramname">options</td><td>The options flag indicating how/where to store the soft constraints </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga73cdc07b9a199c614367bebef0f2c41a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_sc_remove </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Remove soft constraints from <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a>. </p>
<dl class="section note"><dt>Note</dt><dd>Accepts vrna_fold_compound_t of type <a class="el" href="group__fold__compound.html#gga01a4ff86fa71deaaa5d1abbd95a1447da1608d3aa78905fc39e0d25a624ac9512">VRNA_VC_TYPE_SINGLE</a> and <a class="el" href="group__fold__compound.html#gga01a4ff86fa71deaaa5d1abbd95a1447da056345f1bcfe7cd595d1fd437c05246d">VRNA_VC_TYPE_ALIGNMENT</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> possibly containing soft constraints </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga6d55446448d69346fc313b993c4fb3e8"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_sc_free </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__constraints.html#ga75401ce219ef8dbcceb672db82d434c6">vrna_sc_t</a> *&#160;</td>
          <td class="paramname"><em>sc</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Free memory occupied by a <a class="el" href="group__constraints.html#ga75401ce219ef8dbcceb672db82d434c6" title="Typename for the soft constraints data structure vrna_sc_s. ">vrna_sc_t</a> data structure. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">sc</td><td>The data structure to free from memory </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ga57d612b58e1c61dd6cfcb5a843f8f1b3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int vrna_sc_add_SHAPE_deigan </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const double *&#160;</td>
          <td class="paramname"><em>reactivities</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>m</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>b</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Add SHAPE reactivity data as soft constraints (Deigan et al. method) </p>
<p>This approach of SHAPE directed RNA folding uses the simple linear ansatz </p><p class="formulaDsp">
<img class="formulaDsp" alt="\[ \Delta G_{\text{SHAPE}}(i) = m \ln(\text{SHAPE reactivity}(i)+1)+ b \]" src="form_30.png"/>
</p>
<p> to convert SHAPE reactivity values to pseudo energies whenever a nucleotide <img class="formulaInl" alt="$ i $" src="form_2.png"/> contributes to a stacked pair. A positive slope <img class="formulaInl" alt="$ m $" src="form_31.png"/> penalizes high reactivities in paired regions, while a negative intercept <img class="formulaInl" alt="$ b $" src="form_32.png"/> results in a confirmatory ``bonus'' free energy for correctly predicted base pairs. Since the energy evaluation of a base pair stack involves two pairs, the pseudo energies are added for all four contributing nucleotides. Consequently, the energy term is applied twice for pairs inside a helix and only once for pairs adjacent to other structures. For all other loop types the energy model remains unchanged even when the experimental data highly disagrees with a certain motif.</p>
<dl class="section see"><dt>See also</dt><dd>For further details, we refer to <a class="el" href="citelist.html#CITEREF_deigan:2009">[3]</a>. </dd>
<dd>
<a class="el" href="group__soft__constraints.html#ga73cdc07b9a199c614367bebef0f2c41a" title="Remove soft constraints from vrna_fold_compound_t. ">vrna_sc_remove()</a>, <a class="el" href="group__soft__constraints.html#gaf3c65a045060aef5c4e41693d30af58c" title="Add SHAPE reactivity data as soft constraints (Zarringhalam et al. method) ">vrna_sc_add_SHAPE_zarringhalam()</a>, <a class="el" href="group__soft__constraints.html#gaa124bdc20d88001c38ade590c4bcc3c4" title="Find a vector of perturbation energies that minimizes the discripancies between predicted and observe...">vrna_sc_minimize_pertubation()</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> the soft constraints are associated with </td></tr>
    <tr><td class="paramname">reactivities</td><td>A vector of normalized SHAPE reactivities </td></tr>
    <tr><td class="paramname">m</td><td>The slope of the conversion function </td></tr>
    <tr><td class="paramname">b</td><td>The intercept of the conversion function </td></tr>
    <tr><td class="paramname">options</td><td>The options flag indicating how/where to store the soft constraints </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>1 on successful extraction of the method, 0 on errors </dd></dl>

</div>
</div>
<a class="anchor" id="ga04ba85da63d8c793bb8001d1e6f800ba"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int vrna_sc_add_SHAPE_deigan_ali </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char **&#160;</td>
          <td class="paramname"><em>shape_files</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const int *&#160;</td>
          <td class="paramname"><em>shape_file_association</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>m</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>b</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Add SHAPE reactivity data from files as soft constraints for consensus structure prediction (Deigan et al. method) </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> the soft constraints are associated with </td></tr>
    <tr><td class="paramname">shape_files</td><td>A set of filenames that contain normalized SHAPE reactivity data </td></tr>
    <tr><td class="paramname">shape_file_association</td><td>An array of integers that associate the files with sequences in the alignment </td></tr>
    <tr><td class="paramname">m</td><td>The slope of the conversion function </td></tr>
    <tr><td class="paramname">b</td><td>The intercept of the conversion function </td></tr>
    <tr><td class="paramname">options</td><td>The options flag indicating how/where to store the soft constraints </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>1 on successful extraction of the method, 0 on errors </dd></dl>

</div>
</div>
<a class="anchor" id="gaf3c65a045060aef5c4e41693d30af58c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int vrna_sc_add_SHAPE_zarringhalam </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const double *&#160;</td>
          <td class="paramname"><em>reactivities</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>b</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>default_value</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>shape_conversion</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned int&#160;</td>
          <td class="paramname"><em>options</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Add SHAPE reactivity data as soft constraints (Zarringhalam et al. method) </p>
<p>This method first converts the observed SHAPE reactivity of nucleotide <img class="formulaInl" alt="$ i $" src="form_2.png"/> into a probability <img class="formulaInl" alt="$ q_i $" src="form_33.png"/> that position <img class="formulaInl" alt="$ i $" src="form_2.png"/> is unpaired by means of a non-linear map. Then pseudo-energies of the form </p><p class="formulaDsp">
<img class="formulaDsp" alt="\[ \Delta G_{\text{SHAPE}}(x,i) = \beta\ |x_i - q_i| \]" src="form_34.png"/>
</p>
<p> are computed, where <img class="formulaInl" alt="$ x_i=0 $" src="form_35.png"/> if position <img class="formulaInl" alt="$ i $" src="form_2.png"/> is unpaired and <img class="formulaInl" alt="$ x_i=1 $" src="form_36.png"/> if <img class="formulaInl" alt="$ i $" src="form_2.png"/> is paired in a given secondary structure. The parameter <img class="formulaInl" alt="$ \beta $" src="form_37.png"/> serves as scaling factor. The magnitude of discrepancy between prediction and experimental observation is represented by <img class="formulaInl" alt="$ |x_i - q_i| $" src="form_38.png"/>.</p>
<dl class="section see"><dt>See also</dt><dd>For further details, we refer to <a class="el" href="citelist.html#CITEREF_zarringhalam:2012">[18]</a> </dd>
<dd>
<a class="el" href="group__soft__constraints.html#ga73cdc07b9a199c614367bebef0f2c41a" title="Remove soft constraints from vrna_fold_compound_t. ">vrna_sc_remove()</a>, <a class="el" href="group__soft__constraints.html#ga57d612b58e1c61dd6cfcb5a843f8f1b3" title="Add SHAPE reactivity data as soft constraints (Deigan et al. method) ">vrna_sc_add_SHAPE_deigan()</a>, <a class="el" href="group__soft__constraints.html#gaa124bdc20d88001c38ade590c4bcc3c4" title="Find a vector of perturbation energies that minimizes the discripancies between predicted and observe...">vrna_sc_minimize_pertubation()</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>The <a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166" title="Typename for the fold_compound data structure vrna_fc_s. ">vrna_fold_compound_t</a> the soft constraints are associated with </td></tr>
    <tr><td class="paramname">reactivities</td><td>A vector of normalized SHAPE reactivities </td></tr>
    <tr><td class="paramname">b</td><td>The scaling factor <img class="formulaInl" alt="$ \beta $" src="form_37.png"/> of the conversion function </td></tr>
    <tr><td class="paramname">options</td><td>The options flag indicating how/where to store the soft constraints </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>1 on successful extraction of the method, 0 on errors </dd></dl>

</div>
</div>
<a class="anchor" id="ga67675b3ed48744489a3bcfa4174197cb"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int vrna_sc_SHAPE_to_pr </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>shape_conversion</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double *&#160;</td>
          <td class="paramname"><em>values</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>length</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>default_value</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="constraints_8h.html">ViennaRNA/constraints.h</a>&gt;</code></p>

<p>Convert SHAPE reactivity values to probabilities for being unpaired. </p>
<p>This function parses the informations from a given file and stores the result in the preallocated string sequence and the <a class="el" href="group__data__structures.html#ga31125aeace516926bf7f251f759b6126" title="Typename for floating point number in partition function computations. ">FLT_OR_DBL</a> array values.</p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__file__utils.html#ga646ebf45450a69a7f2533f9ecd283a32" title="Read data from a given SHAPE reactivity input file. ">vrna_file_SHAPE_read()</a> </dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">shape_conversion</td><td>String definining the method used for the conversion process </td></tr>
    <tr><td class="paramname">values</td><td>Pointer to an array of SHAPE reactivities </td></tr>
    <tr><td class="paramname">length</td><td>Length of the array of SHAPE reactivities </td></tr>
    <tr><td class="paramname">default_value</td><td>Result used for position with invalid/missing reactivity values </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="gaa124bdc20d88001c38ade590c4bcc3c4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void vrna_sc_minimize_pertubation </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__fold__compound.html#ga1b0cef17fd40466cef5968eaeeff6166">vrna_fold_compound_t</a> *&#160;</td>
          <td class="paramname"><em>vc</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const double *&#160;</td>
          <td class="paramname"><em>q_prob_unpaired</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>objective_function</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>sigma_squared</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>tau_squared</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>algorithm</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>sample_size</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double *&#160;</td>
          <td class="paramname"><em>epsilon</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>initialStepSize</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>minStepSize</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>minImprovement</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>minimizerTolerance</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__soft__constraints.html#gaa715397c7afd2d2955c315512a3d571a">progress_callback</a>&#160;</td>
          <td class="paramname"><em>callback</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;ViennaRNA/perturbation_fold.h&gt;</code></p>

<p>Find a vector of perturbation energies that minimizes the discripancies between predicted and observed pairing probabilities and the amount of neccessary adjustments. </p>
<p>Use an iterative minimization algorithm to find a vector of perturbation energies whose incorporation as soft constraints shifts the predicted pairing probabilities closer to the experimentally observed probabilities. The algorithm aims to minimize an objective function that penalizes discripancies between predicted and observed pairing probabilities and energy model adjustments, i.e. an appropriate vector of perturbation energies satisfies </p><p class="formulaDsp">
<img class="formulaDsp" alt="\[ F(\vec\epsilon) = \sum_{\mu}{ \frac{\epsilon_{\mu}^2}{\tau^2} } + \sum_{i = 1}^n{ \frac{(p_i(\vec\epsilon) - q_i)^2}{\sigma^2} } \to \min. \]" src="form_76.png"/>
</p>
<p>An initialized fold compound and an array containing the observed probability for each nucleotide to be unbound are required as input data. The parameters objective_function, sigma_squared and tau_squared are responsible for adjusting the aim of the objective function. Dependend on which type of objective function is selected, either squared or absolute aberrations are contributing to the objective function. The ratio of the parameters sigma_squared and tau_squared can be used to adjust the algorithm to find a solution either close to the thermodynamic prediction (sigma_squared &gt;&gt; tau_squared) or close to the experimental data (tau_squared &gt;&gt; sigma_squared). The minimization can be performed by makeing use of a custom gradient descent implementation or using one of the minimizing algorithms provided by the GNU Scientific Library. All algorithms require the evaluation of the gradient of the objective function, which includes the evaluation of conditional pairing probabilites. Since an exact evaluation is expensive, the probabilities can also be estimated from sampling by setting an appropriate sample size. The found vector of perturbation energies will be stored in the array epsilon. The progress of the minimization process can be tracked by implementing and passing a callback function.</p>
<dl class="section see"><dt>See also</dt><dd>For further details we refere to <a class="el" href="citelist.html#CITEREF_washietl:2012">[16]</a>.</dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">vc</td><td>Pointer to a fold compound </td></tr>
    <tr><td class="paramname">q_prob_unpaired</td><td>Pointer to an array containing the probability to be unpaired for each nucleotide </td></tr>
    <tr><td class="paramname">objective_function</td><td>The type of objective function to be used (VRNA_OBJECTIVE_FUNCTION_QUADRATIC / VRNA_OBJECTIVE_FUNCTION_LINEAR) </td></tr>
    <tr><td class="paramname">sigma_squared</td><td>A factor used for weighting the objective function. More weight on this factor will lead to a solution close to the null vector. </td></tr>
    <tr><td class="paramname">tau_squared</td><td>A factor used for weighting the objective function. More weight on this factor will lead to a solution close to the data provided in q_prob_unpaired. </td></tr>
    <tr><td class="paramname">algorithm</td><td>The minimization algorithm (VRNA_MINIMIZER_*) </td></tr>
    <tr><td class="paramname">sample_size</td><td>The number of sampled sequences used for estimating the pairing probabilities. A value &lt;= 0 will lead to an exact evaluation. </td></tr>
    <tr><td class="paramname">epsilon</td><td>A pointer to an array used for storing the calculated vector of perturbation energies </td></tr>
    <tr><td class="paramname">callback</td><td>A pointer to a callback function used for reporting the current minimization progress </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Fri Feb 19 2016 20:39:20 for RNAlib-2.2.4 by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.9.1 </li>
  </ul>
</div>
</body>
</html>
